rust/windows-client/src-tauri/src/client.rs

```diff
@@ -139,9 +141,22 @@ pub(crate) fn run() -> Result<()> {
 
 /// `gui::run` but wrapped in `anyhow::Result`
 ///
-/// Error handling can go here soon
+/// Automatically logs or shows error dialogs for important user-actionable errors
 fn run_gui(params: GuiParams) -> Result<()> {
-    Ok(gui::run(params)?)
+    let result = gui::run(params);
+
+    // Make sure errors get logged, at least to stderr
+    if let Err(error) = &result {
+        tracing::error!(?error, "client::run error");
```

Comment:
If you move the logger out of `gui::run` into here, you should be able to still log I think.

Reply:
I can probably do it with an `Arc`, the logger gets passed into some closure and I'd have to share ownership

Reply:
Left a TODO to try that in a future PR